src/Libraries/Microsoft.Extensions.Telemetry/Enrichment/ApplicationEnricherServiceCollectionExtensions.cs

```diff
@@ -19,22 +21,65 @@ public static class ApplicationEnricherServiceCollectionExtensions
-    public static IServiceCollection AddServiceLogEnricher(this IServiceCollection services)
+    [Obsolete(
+        DiagnosticIds.Obsoletions.ObsoleteTelemetryApiMessage,
+        DiagnosticId = DiagnosticIds.Obsoletions.ObsoleteTelemetryApiDiagId,
+        UrlFormat = DiagnosticIds.UrlFormat)]
+    public static IServiceCollection AddServiceLogEnricher(this IServiceCollection services) =>
+        services.AddApplicationLogEnricher(_ => { });
+
+    /// <summary>
+    /// Adds an instance of the service enricher to the <see cref="IServiceCollection"/>.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add the service enricher to.</param>
+    /// <param name="configure">The <see cref="ApplicationLogEnricherOptions"/> configuration delegate.</param>
+    /// <returns>The value of <paramref name="services"/>.</returns>
+    /// <exception cref="ArgumentNullException">Any of the arguments is <see langword="null"/>.</exception>
+    [Obsolete(
+        DiagnosticIds.Obsoletions.ObsoleteTelemetryApiMessage,
+        DiagnosticId = DiagnosticIds.Obsoletions.ObsoleteTelemetryApiDiagId,
+        UrlFormat = DiagnosticIds.UrlFormat)]
+    public static IServiceCollection AddServiceLogEnricher(this IServiceCollection services, Action<ApplicationLogEnricherOptions> configure) =>
+        services.AddApplicationLogEnricher(configure);
+
+    /// <summary>
+    /// Adds an instance of the service enricher to the <see cref="IServiceCollection"/>.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add the service enricher to.</param>
+    /// <param name="section">The <see cref="IConfigurationSection"/> to use for configuring <see cref="ApplicationLogEnricherOptions"/> in the service enricher.</param>
+    /// <returns>The value of <paramref name="services"/>.</returns>
+    /// <exception cref="ArgumentNullException">Any of the arguments is <see langword="null"/>.</exception>
+    [Obsolete(
+        DiagnosticIds.Obsoletions.ObsoleteTelemetryApiMessage,
+        DiagnosticId = DiagnosticIds.Obsoletions.ObsoleteTelemetryApiDiagId,
+        UrlFormat = DiagnosticIds.UrlFormat)]
+    public static IServiceCollection AddServiceLogEnricher(this IServiceCollection services, IConfigurationSection section) =>
+        services.AddApplicationLogEnricher(section);
+
+    /// <summary>
+    /// Adds an instance of the application enricher to the <see cref="IServiceCollection"/>.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add the application enricher to.</param>
+    /// <returns>The value of <paramref name="services"/>.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="services"/> is <see langword="null"/>.</exception>
+    [Experimental(diagnosticId: DiagnosticIds.Experiments.Telemetry, UrlFormat = DiagnosticIds.UrlFormat)]
```

Comment:
Why are we marking experimental to these if we are obsoleting the other ones? IMO we should have a warning free and supported version of the API available, so either keep the new ones as experimental and don't obsolete the old ones, or obsolete the old ones and don't mark these as experimental.